### ISSUE TITLE
REACT: Add option to map variables to variables - Changing the Type of variable (Normal/Derived) in the dataset mapping view

### ISSUE THREAD

**Issue by HayleyMills** [CLOSER-Cohorts/archivist#218]
It is not currently possible to map derived variables to variables in Archivist, as all variables are automatically added as Normal. Restrictions only allow Normal variables to be mapped to questions, and it is not possible to alter the Type to Derived. 

It may be possible to apply the mappings via qv.txt files but this has not yet been tested.

**Comment by HayleyMills** [comment]
DV mappings can be added via .txt files only.

**Comment by HayleyMills** [comment]
The requirement is for the React interface to be able to map variables to variables. Currently this can only be done via uploading dv.txt files. In the original archivist, once a derived variable was recognised (via the txt files) as Derived, you could then continue to to add variables (see example below) this is not the case for the react version, were Derived variables still only have the option of adding questions. Open to ideas as the the best way to do this, but I think having the option to change a variable from Normal to Derived would then determine which drop-down yoy get in the Sources. Please let me know if any of this is not clear. 

![image](https://user-images.githubusercontent.com/16664703/151836716-3062d59f-dde9-485f-a2fc-1d975d66ca31.png)

**Comment by simonreed** [comment]
@HayleyMills this is now on staging, you can select if the variable type is 'Normal' or 'Derived' the sources dropdown will then change to either show questions if Normal or variables if Derived.

<img width="1091" alt="Screenshot 2022-02-04 at 11 16 53" src="https://user-images.githubusercontent.com/13276/152520098-b01ef162-3154-4241-bf16-7090b9a4da10.png">

**Comment by HayleyMills** [comment]
This great. Tested on staing using dataset https://closer-archivist-staging.herokuapp.com/datasets/694

Can we have the inheritance/conflicts work in the same way as questions? So when a variable is assigned to a variable it will inherit the topic that the source variable has? e.g. I mapped pid to e_hidp which has a topic of demographics, so pid should have inherited the topic demographics. See below. 

![image](https://user-images.githubusercontent.com/16664703/152531592-4c5eb52c-aaa1-4220-bda6-db6729449245.png)

Also if I search of hidp it only finds the original e_hidp variable and not the varibale which has a source of e_hidp, this is not the case if for example I search for qc_2 the result show the variables with a source of qc_2.
